instaloader/structures.py

```diff
@@ -583,9 +584,20 @@ def _postcommentanswers(node):
                 'https://www.instagram.com/p/{0}/'.format(self.shortcode),
             )
 
+        def _postcommentlikes(node):
+            if node.get('edge_liked_by', {}).get('count', 0) != 0:
+                yield from NodeIterator(
+                    self._context,
+                    '5f0b1f6281e72053cbc07909c8d154ae',
+                    lambda d: d['data']['comment']['edge_liked_by'],
+                    lambda n: Profile(self._context, n),
+                    {'comment_id': node['id']},
+                    'https://www.instagram.com/p/{0}/'.format(self.shortcode),
+                )
+
         def _postcomment(node):
             return PostComment(*_postcommentanswer(node),
-                               answers=_postcommentanswers(node))
+                               answers=_postcommentanswers(node), likes=_postcommentlikes(node))
```

Comment:
This will trigger an additional GraphQL query for every comment that has a like, even if the likes are not accessed by the user.

One way to solve that problem would be to convert PostComment into a real class with a `@property` `likes()` that does the query only if `.likes` is actually called.

Reply:
Yes totally missed that, thanks for letting me know, Also it's not needed to convert  `PostComment`  into a real class, for `@property` `likes()` to work. But if the code seems awkward or something, i could push a change to convert it into a real class.

Reply:
Indeed I'd rather prefer this PostComment being a real class, considering that a NamedTuple normally simply stores values in a named manner, and our PostComment is now more complex than that. Also it would allow us to 'hide' the `context` as a `_context` like it's done in other structures that need to have the context.

Reply:
I have pushed a commit changing the class, plus other things. It should be good now.